Application/CsrUtil/CsrUtil.c

```diff
@@ -0,0 +1,187 @@
+/** @file
+  Basic reimplementation of SIP aspects of csrutil.
+
+  Copyright (c) 2021, Mike Beaton. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-3-Clause
+**/
+
+
+#include <Uefi.h>
+#include <Library/OcMiscLib.h>
+#include <Library/OcBootManagementLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Library/UefiLib.h>
+#include <IndustryStandard/AppleCsrConfig.h>
+#include <Guid/AppleVariable.h>
+
+STATIC
+VOID
+PrintUsage (
+  )
+{
+  Print (L"usage: csrutil <command>\n");
+  Print (L"Modify the System Integrity Protection configuration.\n");
+  Print (L"Available commands:\n");
+  Print (L"\n");
+  Print (L"    clear\n");
+  Print (L"        Clear the existing configuration.\n");
+  Print (L"    disable\n");
+  Print (L"        Disable the protection on the machine.\n");
+  Print (L"    enable\n");
+  Print (L"        Enable the protection on the machine.\n");
+  Print (L"    status\n");
+  Print (L"        Display the current configuration.\n");
+  Print (L"\n");
+}
+
+EFI_STATUS
+EFIAPI
+UefiMain (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS                            OldStatus;
+  EFI_STATUS                            Status;
+  UINTN                                 Data;
+  UINTN                                 Len;
+  CHAR16                                *EndPtr;
+  UINTN                                 Argc;
+  CHAR16                                **Argv;
+  UINT32                                CsrConfig;
+  UINT32                                Attributes;
+  BOOLEAN                               Usage;
+
+  Status = GetArguments (&Argc, &Argv);
```

Comment:
I think outside of Shell only 1 argument is supported.

Reply:
Oh - though I have not intended this for use outside shell. I was trying to make a thing for inside the shell that looked nicely quite like the Apl tool.

Reply:
@vit9696 - Do I need to make any changes here?

Reply:
Actually, in particular, do I need to confirm this can be called usefully as an entry in `Tools` from OC as well? Have not done yet, but guess it makes sense?

Actually - also relevant - I commented out the link with OC console entry mode lib in the .inf file, because using it makes the screen flash and redraw every time the tool is called from UEFI Shell, which I guess is not so great if its main use case is as a shell tool. Just to double check with you that that decision and reason sounds right?

Reply:
Okay - doing this now!

Reply:
@vit9696 Okay made it work as tool or from command line.